lib/ObjWriter/objwriter.cpp

```diff
@@ -631,6 +734,41 @@ extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
   OST.EmitCVLinetableDirective(OW->FuncId++, Fn, FnEnd);
 }
 
+extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
+                                      const char *FunctionName,
+                                      int FunctionSize) {
+  assert(OW && "ObjWriter is null");
+  auto *AsmPrinter = &OW->getAsmPrinter();
+  auto &OST = static_cast<MCObjectStreamer &>(*AsmPrinter->OutStreamer);
+  MCContext &OutContext = OST.getContext();
+  const MCObjectFileInfo *MOFI = OutContext.getObjectFileInfo();
+
+  if (MOFI->getObjectFileType() == MOFI->IsCOFF) {
+    EmitCVDebugFunctionInfo(OW, FunctionName, FunctionSize);
+  } else {
+    // TODO: Should convert this for non-Windows.
+  }
+}
+
+extern "C" void EmitDebugVar(ObjectWriter *OW, char *name, int typeIndex,
+                             bool isParm, int rangeCount,
+                             ICorDebugInfo::NativeVarInfo *ranges) {
+  assert(OW && "ObjWriter is null");
+
+  if (rangeCount == 0)
+    return;
+
+  int varNumber = ranges[0].varNumber;
```

Comment:
varNumber is just used in assertion. This will fail with release build. Just inline it inside assert below.